### PR TITLE
feat(devtools): support multi jscontext debug

### PR DESCRIPTION
**Pull request by lavnFan**
# Pre-PR Checklist

- [x] I added/updated relevant documentation.
- [x] I followed the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters to submit commit message.
- [x] I squashed the repeated code commits.
- [x] I signed the [CLA].
- [x] I added/updated test cases to check the change I am making.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: android/sdk/src/main/jni/src/bridge/entry.cc [hunk 1]
```diff
@@ -464,16 +465,22 @@ jlong InitInstance(JNIEnv* j_env,
...
-      std::lock_guard<std::mutex> lock(inspector_mutex);
-      if (!global_inspector) {
-        global_inspector = std::make_shared<V8InspectorClientImpl>(scope);
-        global_inspector->Connect(runtime->GetBridge());
-      } else {
-        global_inspector->Reset(scope, runtime->GetBridge());
+      if (runtime->IsDebug()) {
+        std::lock_guard<std::mutex> lock(inspector_mutex);
+        auto inspector = runtime->GetEngine()->GetInspector();
+        if (inspector) {
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/src/bridge/entry.cc:471]: 看起来这里会先于new InspectorClient，这里为空会不会导致第一次没初始化

File: android/sdk/src/main/jni/src/bridge/entry.cc [hunk 2]
```diff
@@ -531,8 +538,21 @@ jlong InitInstance(JNIEnv* j_env,
       engine = std::make_shared<Engine>(std::move(engine_cb_map), param);
       runtime->SetEngine(engine);
       reuse_engine_map[group] = std::make_pair(engine, 1);
+#ifndef V8_WITHOUT_INSPECTOR
+      std::shared_ptr<JavaScriptTask> js_task = std::make_shared<JavaScriptTask>();
+      js_task->callback = [runtime_id]() {
+        std::shared_ptr<Runtime> runtime = Runtime::Find(runtime_id);
+        if (runtime->IsDebug()) {
+          std::lock_guard<std::mutex> lock(inspector_mutex);
+          auto inspector = std::make_shared<V8InspectorClientImpl>();
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/src/bridge/entry.cc:547]: V8InspectorClientImpl为什么不在vm的cb里执行，而要新写个task呢

Comment (lavnFan) [reply android/sdk/src/main/jni/src/bridge/entry.cc]: 可以

File: core/src/inspector/v8_inspector_client_impl.cc [hunk 3]
```diff
@@ -20,34 +20,40 @@
...
   v8::HandleScope handle_scope(isolate);
   inspector_ = v8_inspector::V8Inspector::create(isolate, this);
 }
 
-void V8InspectorClientImpl::Reset(std::shared_ptr<Scope> scope,
-                                  std::shared_ptr<JavaRef> bridge) {
+std::shared_ptr<V8InspectorContext> V8InspectorClientImpl::CreateInspectorContext(std::shared_ptr<Scope> scope, const std::shared_ptr<Bridge>& bridge) {
   scope_ = std::move(scope);
-  channel_->SetBridge(std::move(bridge));
+  auto context_group_id = context_group_count++;
```
[the remark this answers is not in the document]
Comment (lavnFan) [reply core/src/inspector/v8_inspector_client_impl.cc]: 正常外部有加锁还好，这里也改了

File: android/sdk/src/main/jni/src/bridge/bridge.cc [hunk 4]
```diff
@@ -1,41 +1,12 @@
-/*
- *
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/src/bridge/bridge.cc:2]: 版权头怎么删了

File: android/sdk/src/main/jni/src/bridge/entry.cc [hunk 5]
```diff
@@ -442,6 +443,14 @@ jlong InitInstance(JNIEnv* j_env,
       isolate->SetData(kRuntimeSlotIndex, reinterpret_cast<void*>(kReuseRuntimeId));
     }
     isolate->AddMessageListener(HandleUncaughtJsError);
+#ifndef V8_WITHOUT_INSPECTOR
+    std::shared_ptr<Runtime> runtime = Runtime::Find(runtime_id);
+    if (runtime->IsDebug()) {
+      std::lock_guard<std::mutex> lock(inspector_mutex);
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/src/bridge/entry.cc:449]: global_inspector没有了，这里是要锁哪个？

File: android/sdk/src/main/jni/src/bridge/entry.cc [hunk 6]
```diff
@@ -532,7 +540,8 @@ jlong InitInstance(JNIEnv* j_env,
       runtime->SetEngine(engine);
       reuse_engine_map[group] = std::make_pair(engine, 1);
     }
-  } else if (group != kDefaultEngineId) {
+  } else
+    if (group != kDefaultEngineId) {
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/src/bridge/entry.cc:544]: if 格式不太对

File: core/include/core/inspector/v8_inspector_context.h [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+ private:
+  int32_t context_group_id_;
+  std::unique_ptr<V8ChannelImpl> channel_;
+  std::unique_ptr<v8_inspector::V8InspectorSession> session_;
+  std::shared_ptr<Scope> scope_;
+};
+}  // namespace inspector
+}  // namespace hippy
```
Comment (churchill-zhang) [inline core/include/core/inspector/v8_inspector_context.h:59]: 空行

**Review by churchill-zhang — APPROVED** [review]

**Review by medns — APPROVED** [review]
LGTM

**Review by siguangli — APPROVED** [review]

**Review by ozonelmy — APPROVED** [review]

**Pull request merged**